src/utils/uid.js

```diff
@@ -8,7 +8,7 @@ import generate from 'uid'
  */
 
 function uid() {
-  return generate(4)
+  return String(N++)
```

Comment:
No race condition. I guess N must not be initialized to 0. If one loads a persisted slate state into a new slate instance, the state of N is lost and starts creating nodes from 0 again, causing duplicate uids.